crates/router/src/connector/bankofamerica/transformers.rs

```diff
@@ -63,105 +192,290 @@ impl TryFrom<&BankofamericaRouterData<&types::PaymentsAuthorizeRouterData>>
+                    Err(_) => None,
                 };
+                let payment_information = PaymentInformation {
+                    card: Card {
+                        number: ccard.card_number,
+                        expiration_month: ccard.card_exp_month,
+                        expiration_year: ccard.card_exp_year,
+                        security_code: ccard.card_cvc,
+                        card_type,
+                    },
+                };
+
+                let processing_information = ProcessingInformation {
+                    capture: matches!(
+                        item.router_data.request.capture_method,
+                        Some(enums::CaptureMethod::Automatic) | None
+                    ),
+                    capture_options: None,
+                };
+
+                let client_reference_information = ClientReferenceInformation {
+                    code: Some(item.router_data.connector_request_reference_id.clone()),
+                };
+
                 Ok(Self {
-                    amount: item.amount.to_owned(),
-                    card,
+                    processing_information,
+                    payment_information,
+                    order_information,
+                    client_reference_information,
                 })
             }
             _ => Err(errors::ConnectorError::NotImplemented("Payment methods".to_string()).into()),
         }
     }
 }
 
-//TODO: Fill the struct with respective fields
-// Auth Struct
-pub struct BankofamericaAuthType {
-    pub(super) api_key: Secret<String>,
-}
-
-impl TryFrom<&types::ConnectorAuthType> for BankofamericaAuthType {
-    type Error = error_stack::Report<errors::ConnectorError>;
-    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
-        match auth_type {
-            types::ConnectorAuthType::HeaderKey { api_key } => Ok(Self {
-                api_key: api_key.to_owned(),
-            }),
-            _ => Err(errors::ConnectorError::FailedToObtainAuthType.into()),
-        }
-    }
-}
-// PaymentsResponse
-//TODO: Append the remaining status flags
-#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
-#[serde(rename_all = "lowercase")]
+#[derive(Debug, Default, Clone, Deserialize, Eq, PartialEq)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
 pub enum BankofamericaPaymentStatus {
+    Authorized,
     Succeeded,
     Failed,
+    Voided,
+    Reversed,
+    Pending,
+    Declined,
+    AuthorizedPendingReview,
+    Transmitted,
     #[default]
     Processing,
 }
 
 impl From<BankofamericaPaymentStatus> for enums::AttemptStatus {
     fn from(item: BankofamericaPaymentStatus) -> Self {
         match item {
-            BankofamericaPaymentStatus::Succeeded => Self::Charged,
-            BankofamericaPaymentStatus::Failed => Self::Failure,
+            BankofamericaPaymentStatus::Authorized
+            | BankofamericaPaymentStatus::AuthorizedPendingReview => Self::Authorized,
```

Comment:
what is the definition of `AuthorizedPendingReview`, why is this mapped to `Authorized`?

Reply:
Done

Reply:
https://support.cybersource.com/knowledgebase/Knowledgearticle/?code=000002478

Reply:
This will be addressed in separate issue https://github.com/juspay/hyperswitch-cloud/issues/3226